src/util/isArrayLike.ts

```diff
@@ -0,0 +1 @@
+export const isArrayLike = ((x: any): boolean => x && typeof x.length === 'number');
```

Comment:
You can make this a type guard, so TS can better infer types inside the if block;

    export const isArrayLike = (<T>(x: any): x is ArrayLike<T> => x && typeof x.length === 'number');